langwatch/src/server/background/workers/evaluationsWorker.ts

```diff
@@ -95,14 +101,64 @@ export const runEvaluationForTrace = async ({
     };
   }
 
-  const input = trace.input?.value;
-  const output = trace.output?.value;
-  const expected_output = trace.expected_output?.value;
+  const inputMapping = (mappings ?? {})?.input;
+  const outputMapping = (mappings ?? {})?.output;
+  const ragContextMapping = (mappings ?? {})?.context;
+  const expectedOutputMapping = (mappings ?? {})?.expected_output;
+
+  const switchMapping = (mapping: Mappings) => {
+    if (mapping === "trace.input") {
+      return trace.input?.value;
+    }
+    if (mapping === "trace.output") {
+      return trace.output?.value;
+    }
+    if (mapping === "trace.first_rag_context") {
+      const ragInfo = getRAGInfo(spans);
+      return ragInfo.contexts[0];
+    }
+    if (mapping === "metadata.expected_output") {
+      return trace.expected_output?.value;
```

Comment:
can we add a `?? trace.metadata.expected_output` to it too? This way both if it's at the root level at the trace, or inside the metadata as the name suggests, it will work